File: back-end/src/main/java/com/huyphan/models/NewPost.java
package com.huyphan.models;

import com.huyphan.models.enums.PostContentType;
import lombok.Getter;
import lombok.Setter;

/**
 * Information required to add new post.
 */
@Getter
@Setter
public class NewPost {

    private String title;

    private String mediaUrl;

    private String mediaType;

    private Section section;

    private String tags;

    private String text;

    private PostContentType contentType;

    private boolean anonymous;

    private boolean notificationEnabled;
}

File: client/src/models/new-post.ts
import { PostContentType } from './enums/post-content-type';
import Section from './section';

export default interface NewPost {
  readonly title?: string;

  readonly mediaUrl: string | null;

  readonly mediaType: string | null;

  readonly section: Section;

  readonly tags: string;

  readonly text: string | null;

  readonly contentType: PostContentType;

  readonly anonymous: boolean;

  readonly notificationEnabled: boolean;

  readonly nsfw: boolean;

  readonly moderating: boolean;
}
